include/nbl/asset/IImage.h

```diff
@@ -232,9 +238,10 @@ class IImage : public IDescriptor
 			E_SAMPLE_COUNT_FLAGS						samples;
 			// stuff below is irrelevant in OpenGL backend
 			E_TILING									tiling = static_cast<E_TILING>(0);
-			std::underlying_type_t<E_USAGE_FLAGS>		usage = 0u;
+			core::bitflag<E_USAGE_FLAGS>				usage = static_cast<E_USAGE_FLAGS>(0);
 			E_SHARING_MODE								sharingMode = ESM_EXCLUSIVE;
-			core::smart_refctd_dynamic_array<uint32_t>	queueFamilyIndices = nullptr;
+			uint32_t									queueFamilyIndexCount;
+			const uint32_t*								queueFamilyIndices;
```

Comment:
@achalpandeyy & @Crisspl what is this stuff doing here!?

Reply:
This comment https://github.com/Devsh-Graphics-Programming/Nabla/pull/160/files/6634885ebe66f4e9ba1657631f0902e0773ff7ba#r691105872

Reply:
> 
> 
> i thought you guys established something and this is new way of specifying queue fam indices

yeah that's fine, I have no clue what this is doing in the base of ICPUImage

same thing as with ICPUBuffer 

assign @achalpandeyy , this needs to move IGPUImage

in ICPUBuffer and ICPUImage you only need to track core::bitflag<E_USAGE_FLAGS> m_usage;

Reply:
i think matt means that SCreationParams should be in IGPUBuffer, not in IBuffer (common base) since queue indices have nothing to do with asset namespace. Same goes for IImage

Reply:
Yea moving it to IGPUBuffer will work and I'll take care of the usage flags part.

Reply:
ok but then also take care of the IImage problem